internal/direct.go

```diff
@@ -21,50 +22,96 @@ const (
 	authzHeader     = "Authorization"
 	bearer          = "Bearer "
 	gzipFormat      = "gzip"
-	octetStream     = "application/octet-stream"
-	reportEndpoint  = "/report"
 	formatKey       = "f"
 )
 
 // The implementation of a Reporter that reports points directly to a Wavefront server.
 type directReporter struct {
-	serverURL string
-	token     string
+	serverURL   string
+	token       string
+	endpoint    string
+	contentType string
+	gzip        bool
 }
 
-func NewDirectReporter(server string, token string) Reporter {
-	return &directReporter{serverURL: server, token: token}
+// ReporterOption allows Reporter configuration
+type ReporterOption func(*directReporter)
+
+// SetEndpoint set the Reporter endpoint, '/report' by default
+func SetEndpoint(endpoint string) ReporterOption {
+	return func(reporter *directReporter) {
+		reporter.endpoint = endpoint
+	}
+}
+
+// SetContentType set the Reporter contentType, 'application/octet-stream' by default
+func SetContentType(contentType string) ReporterOption {
+	return func(reporter *directReporter) {
+		reporter.contentType = contentType
+	}
+}
+
+// EnableGZip set the Reporter gzip compression, 'true' by default
+func EnableGZip(gzip bool) ReporterOption {
+	return func(reporter *directReporter) {
+		reporter.gzip = gzip
+	}
+}
+
+// NewDirectReporter create a Reporter
+func NewDirectReporter(server string, token string, setters ...ReporterOption) Reporter {
+	dr := &directReporter{
+		serverURL:   server,
+		token:       token,
+		endpoint:    "/report",
+		contentType: "application/octet-stream",
+		gzip:        true,
+	}
+	for _, setter := range setters {
+		setter(dr)
+	}
+	return dr
 }
 
 func (reporter directReporter) Report(format string, pointLines string) (*http.Response, error) {
```

Comment:
It'd be cleaner to have a separate method for reporting events and invoke that from the LineHandler based on the format. The `Report` method was really meant to handle the `/report?f=` endpoint. Let's leave that as is and add a new one for reporting events.